lib/object.js

```diff
@@ -199,28 +200,28 @@ function defaults (opts) {
 
       switch (f.type) {
         case schema.STRING:
-          gen(`${f.name}: ${JSON.stringify(f.default || '')}${s}`)
+          gen(`['${f.name}']: ${JSON.stringify(f.default || '')}${s}`)
           break
         case schema.NUMBER:
-          gen(`${f.name}: ${f.default || 0}${s}`)
+          gen(`['${f.name}']: ${f.default || 0}${s}`)
           break
         case schema.BOOLEAN:
-          gen(`${f.name}: ${f.default || false}${s}`)
+          gen(`['${f.name}']: ${f.default || false}${s}`)
           break
         case schema.ARRAY:
           if (isRequired(f)) {
-            gen(`${f.name}: []${s}`)
+            gen(`['${f.name}']: []${s}`)
           } else {
-            gen(`${f.name}: undefined${s}`)
+            gen(`['${f.name}']: undefined${s}`)
           }
           break
         case schema.OBJECT:
           if (isRequired(f)) {
-            gen(`${f.name}: {`)
+            gen(`['${f.name}']: {`)
             defaultFields(gen, f.fields)
             gen(`}${s}`)
           } else {
-            gen(`${f.name}: undefined${s}`)
+            gen(`['${f.name}']: undefined${s}`)
           }
           break
```

Comment:
You can use this module, https://github.com/mikolalysenko/is-property to determine is a prop needs to be escaped using the [ ] syntax. If not it's gonna be faster to stick with what we have :)

Also instead of doing `'${f.name}'` do `${JSON.stringify(f.name)}` as that fixes any escapeing etc

Reply:
Actually use this instead, https://github.com/mafintosh/generate-object-property - it has a .property method that does exactly what we need